### ISSUE TITLE
Oracle: PL/SQL control structures (IF/ELSIF/ELSE/FOR) lack proper indentation rules

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7057]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff lacks proper indentation rules for Oracle PL/SQL control structures including IF/ELSIF/ELSE blocks and FOR loops, resulting in inconsistent or incorrect formatting.

### Expected Behaviour

Oracle PL/SQL control structures should be properly indented:

```sql
BEGIN
  IF 1 > 0 THEN
    IF 10 > 5 THEN
      DBMS_OUTPUT.PUT_LINE('Condition met: 10 > 5');
    ELSE
      DBMS_OUTPUT.PUT_LINE('Condition not met');
    END IF;

    IF 'ACTIVE' = 'ACTIVE' THEN
      DBMS_OUTPUT.PUT_LINE('Status is ACTIVE');
    ELSIF 'ACTIVE' = 'INACTIVE' THEN
      DBMS_OUTPUT.PUT_LINE('Status is INACTIVE');
    ELSE
      DBMS_OUTPUT.PUT_LINE('Status is unknown');
    END IF;

    IF 'A' = 'A' THEN
      DBMS_OUTPUT.PUT_LINE('Grade A detected');
    ELSIF 'A' = 'B' THEN
      DBMS_OUTPUT.PUT_LINE('Grade B detected');
    ELSIF 'A' = 'C' THEN
      DBMS_OUTPUT.PUT_LINE('Grade C detected');
    ELSE
      DBMS_OUTPUT.PUT_LINE('Unknown grade');
    END IF;

  ELSE
    DBMS_OUTPUT.PUT_LINE('Initial condition failed');
  END IF;

  FOR emp_rec IN (
    SELECT
      employee_id,
      first_name,
      last_name,
      salary,
      department_id
    FROM employees
    WHERE department_id = 10
  )
  LOOP
    IF emp_rec.salary > 50000 THEN
      DBMS_OUTPUT.PUT_LINE('High earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
    ELSIF emp_rec.salary > 30000 THEN
      DBMS_OUTPUT.PUT_LINE('Medium earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
    ELSE
      DBMS_OUTPUT.PUT_LINE('Standard earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
    END IF;
  END LOOP;
END;
```

### Observed Behaviour

```sql
BEGIN
  IF 1 > 0 THEN
  IF 10 > 5 THEN
  DBMS_OUTPUT.PUT_LINE('Condition met: 10 > 5');
  ELSE
  DBMS_OUTPUT.PUT_LINE('Condition not met');
  END IF;

  IF 'ACTIVE' = 'ACTIVE' THEN
  DBMS_OUTPUT.PUT_LINE('Status is ACTIVE');
  ELSIF 'ACTIVE' = 'INACTIVE' THEN
  DBMS_OUTPUT.PUT_LINE('Status is INACTIVE');
  ELSE
  DBMS_OUTPUT.PUT_LINE('Status is unknown');
  END IF;

  IF 'A' = 'A' THEN
  DBMS_OUTPUT.PUT_LINE('Grade A detected');
  ELSIF 'A' = 'B' THEN
  DBMS_OUTPUT.PUT_LINE('Grade B detected');
  ELSIF 'A' = 'C' THEN
  DBMS_OUTPUT.PUT_LINE('Grade C detected');
  ELSE
  DBMS_OUTPUT.PUT_LINE('Unknown grade');
  END IF;

  ELSE
  DBMS_OUTPUT.PUT_LINE('Initial condition failed');
  END IF;

  FOR emp_rec IN (
    SELECT
      employee_id,
      first_name,
      last_name,
      salary,
      department_id
    FROM employees
    WHERE department_id = 10
  )
  LOOP
  IF emp_rec.salary > 50000 THEN
  DBMS_OUTPUT.PUT_LINE('High earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
  ELSIF emp_rec.salary > 30000 THEN
  DBMS_OUTPUT.PUT_LINE('Medium earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
  ELSE
  DBMS_OUTPUT.PUT_LINE('Standard earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
  END IF;
  END LOOP;
END;
```

### How to reproduce

1. Create an Oracle PL/SQL file (e.g., plsql_indent.sql) with the following content:

```sql
BEGIN
IF 1 > 0 THEN
IF 10 > 5 THEN
DBMS_OUTPUT.PUT_LINE('Condition met: 10 > 5');
ELSE
DBMS_OUTPUT.PUT_LINE('Condition not met');
END IF;

IF 'ACTIVE' = 'ACTIVE' THEN
DBMS_OUTPUT.PUT_LINE('Status is ACTIVE');
ELSIF 'ACTIVE' = 'INACTIVE' THEN
DBMS_OUTPUT.PUT_LINE('Status is INACTIVE');
ELSE
DBMS_OUTPUT.PUT_LINE('Status is unknown');
END IF;

IF 'A' = 'A' THEN
DBMS_OUTPUT.PUT_LINE('Grade A detected');
ELSIF 'A' = 'B' THEN
DBMS_OUTPUT.PUT_LINE('Grade B detected');
ELSIF 'A' = 'C' THEN
DBMS_OUTPUT.PUT_LINE('Grade C detected');
ELSE
DBMS_OUTPUT.PUT_LINE('Unknown grade');
END IF;

ELSE
DBMS_OUTPUT.PUT_LINE('Initial condition failed');
END IF;

FOR emp_rec IN (
SELECT
employee_id,
first_name,
last_name,
salary,
department_id
FROM employees
WHERE department_id = 10
)
LOOP
IF emp_rec.salary > 50000 THEN
DBMS_OUTPUT.PUT_LINE('High earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
ELSIF emp_rec.salary > 30000 THEN
DBMS_OUTPUT.PUT_LINE('Medium earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
ELSE
DBMS_OUTPUT.PUT_LINE('Standard earner: ' || emp_rec.first_name || ' ' || emp_rec.last_name);
END IF;
END LOOP;
END;
```

2. Configure sqlfluff for Oracle dialect with indentation rules (see config section).
3. Run sqlfluff to format the file:

```sh
sqlfluff fix plsql_indent.sql
```

### Dialect

oracle

### Version

3.4.2

### Configuration

```ini
[sqlfluff]
dialect = oracle

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 4
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
